ocaml/xapi-cli-server/cli_operations.ml

```diff
@@ -4400,43 +4409,47 @@ let vm_migrate printer rpc session_id params =
              and among the choices of that the shared is preferred first(as it is recommended to have shared storage
              in pool to host VMs), and then the one with the maximum available space *)
           try
-            let query =
-              Printf.sprintf
-                {|(field "host"="%s") and (field "currently_attached"="true")|}
-                (Ref.string_of host)
-            in
-            let host_pbds =
-              Client.PBD.get_all_records_where remote_rpc remote_session query
+            let pbd_in_host self =
```

Comment:
The code below can again lead to lots and lots of calls unfortunately. The `preferred_sr` is only used if the vdi_map is absent or incomplete. I wonder if we can make this more efficient.

Reply:
I've restricted the number of PBDs fetched to the ones present in the host, instead of fetching all the ones in the pool and filtering afterward. I'm trying to come up with other ways to reduce the number of calls